USER_MANUAL.md

```diff
@@ -1,12 +1,355 @@
+        potential ambiguous operators
+
+2.  sentences that define our *system's state*, including:
+
+    -   **configuration declarations:** define labelled, hierarchical records
+        using an nested XML-like syntax
+
+3.  sentences that define our *system's behavior*, including:
+
+    -   **context declarations:** describe how primitives and configurations
+        can simplify
+    -   **context alias declarations:** define templates that can generate new
+        contexts
+    -   **rule declarations:** define how the system transitions from one state
+        to the next
+
+### K Syntax Overview
+
+Here we provide an overview of the top-level K syntax using a K-style BNF
+notation. Note that this is not a full definition; we focus instead on the
+overall structure of the specification. Syntactic categories which are not
+defined are denoted by names surrounded by curly braces (`{}`). In particular,
+we omit the syntax of strings (`{String}`) and all types of identifiers, e.g.,
+`{ModuleId}`, `{Id}`, `{SortId}`, `{CellId}`, etc. We also omit the syntax of
+attributes (`{Attribute}`), a highly flexible mechanism for modifying the
+interpretation of modules and various kinds of sentences.
+
+Due to the highly generic nature of user-defined K syntax, the parsing process
+of K specifications roughly proceeds in three stages, corresponding to the
+three kinds of declarations shown above.
+
+1.  In the first stage, system primitive declarations are gathered and a parser
+    is constructed.
+2.  In the second stage, the system primitive parser is used to parse
+    configuration declarations; in our BNF grammar below, we use `{TermBubble}`
+    to describe things which can be parsed using the primitive syntax.
+3.  Finally, a third stage parser is constructed containing both primitive and
+    configuration syntax is constructed which is used to parse contexts,
+    context aliases, and rules. In our BNF grammar below, we use the words
+    `{Bubble}` and `{BubbleWithRewrites}` to describe parsers which contain the
+    combined primitive and configuration syntax. The `{BubbleWithRewrites}`
+    category additional adds the rewrite arrow syntax (`=>`) needed for parsing
+    `rule`s and `context`s.
+
+**Note:** _the parsers listed above are all used internally by the K compiler
+`kompile`. As part of compilation, `kompile` will generate a separate
+language-specific parser that can only parse programs in the specified language
+syntax. The language-specific parser is the one invoked when using `krun`._
+
+We now present our high-level BNF syntax:
+
+```
+  File         ::= RequiresList ModuleList
+
+  RequiresList ::= "" | Requires RequiresList
+  Requires     ::= RequireCmd {String}
+  RequireCmd   ::= "requires" | "require"
+
+  Module       ::= "module" {ModuleId} AttributeSet ImportList SentenceList "endmodule"
+
+  AttributeSet ::= "" | "[" Attributes "]"
+  Attributes   ::= Attributes "," {Attribute}
```

Comment:
this is not correct as your grammar here does not allow for the possibility of a sentence with only a single attribute.

Reply:
I removed this section from this PR because it needs more polish.